RoddyCore/src/de/dkfz/roddy/config/converters/JSONConverter.groovy

```diff
@@ -0,0 +1,214 @@
+ *
+ * Distributed under the MIT License (license terms are at https://www.github.com/TheRoddyWMS/Roddy/LICENSE.txt).
+ */
+
+package de.dkfz.roddy.config.converters
+
+import de.dkfz.roddy.config.Configuration
+import de.dkfz.roddy.config.ConfigurationValue
+import de.dkfz.roddy.core.ExecutionContext
+import groovy.json.JsonSlurper
+import groovy.transform.CompileStatic
+import groovy.transform.TypeCheckingMode
+import groovy.xml.MarkupBuilder
+import sun.reflect.generics.reflectiveObjects.NotImplementedException
+
+/**
+ * In exception to our other Groovy classes, this class needs to be CompileDynamic!
+ * We use MarkupBuilder closures, where you cannot attach the CompileDynamic directive.
+ *
+ * Created by heinold on 30.01.17.
+ */
+class JSONConverter extends ConfigurationConverter {
+
+    @CompileStatic
+    @Override
+    String convert(ExecutionContext context, Configuration configuration) {
+        throw new NotImplementedException()
+    }
+
+    @CompileStatic
+    @Override
+    StringBuilder convertConfigurationValue(ConfigurationValue cv, ExecutionContext context) {
+        throw new NotImplementedException()
+    }
+
+    @CompileStatic
+    @Override
+    String convertToXML(File file) {
+        convertJsonString(file.text)
+    }
+
+    @CompileStatic
+    private String readAttribute(String attribute, def nc) { nc?.getAt(attribute) ?: "" }
+
+    @CompileStatic
+    private String idOf(def nc) { return readAttribute("id", nc) }
+
+    @CompileStatic
+    private String nameOf(def nc) { return readAttribute("name", nc) }
+
+    @CompileStatic
+    private String valueOf(def nc) { return readAttribute("value", nc) }
+
+    @CompileStatic
+    private String descriptionOf(def nc) { return readAttribute("description", nc) }
+
+    @CompileStatic
+    private String classOf(def nc) { return readAttribute("class", nc) }
+
+    @CompileStatic
+    private String typeOf(def nc) { return readAttribute("type", nc) }
+
+    @CompileStatic
+    private String typeofOf(def nc) { return readAttribute("typeof", nc) }
+
+    @CompileStatic
+    private String scriptparameterOf(def nc) { return readAttribute("scriptparameter", nc) }
+
+    @CompileStatic
+    private String sizeOf(def nc) { return readAttribute("size", nc) }
+
+    @CompileStatic
+    private String memoryOf(def nc) { return readAttribute("memory", nc) }
+
+    @CompileStatic
+    private String coresOf(def nc) { return readAttribute("cores", nc) }
+
+    @CompileStatic
+    private String nodesOf(def nc) { return readAttribute("nodes", nc) }
+
+    @CompileStatic
+    private String walltimeOf(def nc) { return readAttribute("walltime", nc) }
+
+    @CompileStatic(TypeCheckingMode.SKIP)
+    String convertJsonString(String json) {
+        /**
+         Remove comments first, they are not allowed in JSON and will produce errors.
+
+         See also a comment from Douglas Crockford:
```

Comment:
Really, this is just an opinion. This is the same argument like "Perl is bad, because you can write crappy hacks." I would not cite such an opinion in our code. The first comment line is sufficient and the solution to just ignore comments is fine -- although if you want to ignore comments (i.e. allow them in the JSON file), it is good to cite the source of he convention.

Actually, the inflation of braces and quotes and the lack of comments are exactly the reasons no to use JSON and prefer to the more powerful and expressive YAML (not that there are no arguments against YAML either).

Reply:
This statement is from the Json inventor. I don't like that he did it but I understand why. That's why I cited it. But I follow his idea and put in comments which will be removed. Actually I only need them for copyright headers.